src/common/misc/DebugStream.h

```diff
@@ -35,11 +35,11 @@ namespace DebugStream
     extern MISC_API int GetLevel();
 }
 
-#define debug1 if (!DebugStream::Level1()) ; else (DebugStream::Stream1((char const *)__FILE__,(int)__LINE__))
```

Comment:
The problem with this old code is in situations like this...

```
if (myvar > tol)
    debug2 << "variable exceeded tolerance" << endl;
```
gets replaced to...
```
if (myvar > tol)
   if (!DebugStream::Level2())
      ;
   else
      DebugStream::Stream2() << "variable exceeded tolerance" << endl;
```
and the compiler generates warnings about that `else` being *visually* ambiguous.

Reply:
Q: we use this negative case to preserve line #s?

Reply:
sorry -- did we use?

Reply:
regardless, new case looks much better

Reply:
A little more info. The `debug1`...`debug5` macros present to the developer what appears to be an ordinary stream. But, they are macros and embedded within those macros is the logic to determine if the particular debug level is even active. So, rather than *inserting* into a stream object that then asks whether those inserts should proceed, the logic works to not even attempt the inserts if the stream is not active (which as an aside means that any work involved in producing the *stuff* that gets inserted can be skipped). The old `if !active do nothing; else <<` approach lead to compiler warnings in some circumstances. The new `active && <<` approach is just better and it has the same behavior that insert *work* won't happen if the stream is not active.

With respect to the file/line number...we enabled that only for debug stream 1. It does lead to logs with much longer lines (because often the paths to VisIt sources may be > 100 chars).